lca-cli/ibi-preparation/ibipreparation.go

```diff
@@ -211,3 +235,45 @@ func (i *IBIPrepare) cleanupRhcosSysroot() error {
 	}
 	return nil
 }
+
+func (i *IBIPrepare) shouldOverrideSeedRegistry(seedMirrorRegistryConfigured bool, seedReleaseRegistry string) (bool, error) {
+	targetMirroredRegistries, err := mirrorRegistrySourceRegistries()
+	if err != nil {
+		return false, err
+	}
+
+	i.log.Info("Mirror registry prefixes from file: ", targetMirroredRegistries)
+
+	targetMirrorRegistryConfigured := len(targetMirroredRegistries) > 0
+
+	// if the target SNO doesn't have mirror registry configured:
+	//   - and seed SNO has mirror registry configured then we should try to override the registry
+	//   - and seed SNO has no mirror registry configured then we shouldn't try to override the registry
+	if !targetMirrorRegistryConfigured {
+		return seedMirrorRegistryConfigured, nil
+	}
+
+	return !lo.Contains(targetMirroredRegistries, seedReleaseRegistry), nil
+}
+
+func mirrorRegistrySourceRegistries() ([]string, error) {
```

Comment:
https://github.com/openshift-kni/lifecycle-agent/blob/e8a8907961f834444fc215a9ea19e5015179f0c5/utils/client_helper.go#L291

Reply:
Same as https://github.com/openshift-kni/lifecycle-agent/pull/727/files#r1935308620.

Reply:
Probably we can change name to show that it is read from registry conf file?

Reply:
Absolutely, I've updated the function to make it clearer that it reads the registry conf file. PTAL